File: Frontend/src/types/message.ts
export type MessageStatus =
    | "Sending"
    | "Sent"
    | "Delivered"
    | "Read"
    | "Undelivered";

export interface Message {
    id: string;
    conversationId: string;
    text: string;
    senderId: string;
    sentAt: string;
    status: MessageStatus;
}

File: Backend/WaveMessenger.Domain/Entities/Message.cs
namespace WaveMessenger.Domain.Entities;

public class Message
{
    public Guid Id { get; set; }

    public Guid ConversationId { get; set; }

    public Guid SenderId { get; set; }

    public string Text { get; set; } = string.Empty;

    public DateTime SentAt { get; set; }

    public MessageStatus Status { get; set; }
}